sllm_store/csrc/sllm_store/checkpoint_store.cpp

```diff
@@ -378,16 +455,60 @@ MemPtrListMap CheckpointStore::GetDevicePtrsFromMemHandles(
 
 int CheckpointStore::AllocateModelMemory(const std::shared_ptr<Model>& model) {
   if (use_shm_) {
-    if (shared_memory_pool_ == nullptr) {
+    if (shared_memory_pools_[model->GetModelPath()] == nullptr) {
       LOG(ERROR) << "Shared memory pool is not initialized";
       return -1;
     }
-    return model->AllocatePinnedMemory(shared_memory_pool_);
+    return model->AllocatePinnedMemory(
+        shared_memory_pools_[model->GetModelPath()]);
   } else {
     if (memory_pool_ == nullptr) {
       LOG(ERROR) << "Memory pool is not initialized";
       return -1;
     }
     return model->AllocatePinnedMemory(memory_pool_);
   }
-}
\ No newline at end of file
+}
+
+std::unordered_map<int, void*> CheckpointStore::AllocateSharedMemory(
```

Comment:
Once you have the chunk size
1. calculate total memory size needed
2. round up by chunk size, and get number of chunks
3. each handle is a chunk with common prefix
4. AllocateSharedMemory should be a standalone API, since this is called on client side, you can just `new SharedMemoryAllocator` on the fly, so lifetime is until client exit